app/components/SessionExpiryHandler.tsx

```diff
@@ -62,10 +62,11 @@ const SessionExpiryHandler: React.FC = () => {
         <SessionTimeoutHandler
           modalDisplaySecondsBeforeLogout={120}
           onSessionExpired={handleSessionExpired}
+          resetOnChange={[router]}
```

Comment:
This listens to changes in next router and refreshes the session. This alone *should* significantly mitigate the issue 🤞